### PR TITLE
LibJS: Some scoping and related fixes

### PR DESCRIPTION
**Pull request by davidot**
Also now throwing in test-js at the top level gets you a nice stack trace, but inside a describe or test doesn't. :yakbait: anyone?

Fixes 1 annexB test262 test.
Order verified against: v8 and sm

### CODE REVIEW

File: Userland/Libraries/LibJS/Tests/syntax/function-hoisting.js [hunk 1]
```diff
@@ -0,0 +1,352 @@
+const noHoistLexTL = false;
+let canCallNonHoisted = 0;
+
+expect(basicHoistTL()).toEqual("basicHoistTL");
+
+function basicHoistTL() {
```
Comment (linusg) [inline Userland/Libraries/LibJS/Tests/syntax/function-hoisting.js:6]: `TL` -> `TopLevel` (?)

File: Userland/Libraries/LibJS/Tests/syntax/function-hoisting.js [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+        }
+
+        expect(hoistSecondOneFunc()).toEqual("hoistSecondOneFunc");
+    }
+
+    expect(hoistSecondOneFunc()).toEqual("hoistSecondOneFunc");
+
+    expect(notBlockSFTL()).toBe("second");
+
+    function notBlockSFTL() {
```
Comment (linusg) [inline Userland/Libraries/LibJS/Tests/syntax/function-hoisting.js:180]: Please get rid of these acronyms everywhere, they don't exactly help with code readability :)

File: Userland/Libraries/LibJS/Tests/syntax/functions-in-tree-order-strict.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
+"use strict";
+// Note: This must be checked at script level because that is the only place
+//       where function order is visible. We introduce some test(...) with
+//       names to make sure the file does have some tests and a suite.
+
+// Note: In strict mode we expect almost the same result except that the
+// functions in blocks do not get hoisted up.
+// And only the indirect eval copy gets the global var environment.
```
Comment (linusg) [inline Userland/Libraries/LibJS/Tests/syntax/functions-in-tree-order-strict.js:8]: Either indent both to the `Note` level or neither, but this is weird :D

**Comment by davidot** [comment]
E x p a n d e d the acronyms

**Pull request merged**